module/zfs/zfs_vnops.c

```diff
@@ -937,6 +929,8 @@ zfs_write(struct inode *ip, uio_t *uio, int ioflag, cred_t *cr)
 	    zfsvfs->z_os->os_sync == ZFS_SYNC_ALWAYS)
 		zil_commit(zilog, zp->z_id);
 
+	dataset_kstats_update_write_kstats(&zfsvfs->z_kstat,
+	    start_resid - uio->uio_resid);
```

Comment:
In addition to the per-objset IO accounting this PR is adding there is already exists per-task IO accounting.  The calls for this are `task_io_account_read()` and `task_io_account_write()` in `zpl_read_common_iovec()` and `zpl_write_common_iovec()` respectively.  As an aside this, is where the `iotop(8)` is getting its data from.

In the interests of keeping all of the IO accounting in one place let's move these calls in to `zfs_write()` and `zfs_read()`.  This will also have the effect that non-ZPL consumers, like ZVOLS and Lustre servers will get the IO properly accounted to the issuing process.

Reply:
On second thought we probably should leave the `task_io_*` calls where they are or we'll end up double accounting volumes since this is already handled by `submit_bio()`.